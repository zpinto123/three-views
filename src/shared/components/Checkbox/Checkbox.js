import React from "react";
import PropTypes from "prop-types";
import { Checkbox as MUICheckbox } from "@material-ui/core";

const Checkbox = ({ handleChange, color, ...restOfProps }) => (
  <MUICheckbox onChange={handleChange} color={color} {...restOfProps} />
);

Checkbox.propTypes = {
  color: PropTypes.string,
  handleChange: PropTypes.func.isRequired
};

Checkbox.defaultProps = {
  color: "primary"
};

export default Checkbox;
