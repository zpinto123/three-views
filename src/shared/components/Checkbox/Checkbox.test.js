import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import "jest-styled-components";

import Checkbox from "./";

describe("<Checkbox />", () => {
  it("Show Checkbox correctly", () => {
    const mockCallBack = jest.fn();

    const component = renderer.create(<Checkbox handleChange={mockCallBack} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("onChange works correcly", () => {
    const mockCallBack = jest.fn();

    const component = mount(<Checkbox handleChange={mockCallBack} />);
    component.find("input").simulate("change");
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
