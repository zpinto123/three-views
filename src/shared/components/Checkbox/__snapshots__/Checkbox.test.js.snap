// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> Show Checkbox correctly 1`] = `
<span
  aria-disabled={false}
  className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-7 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={null}
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
      />
    </svg>
    <input
      className="PrivateSwitchBase-input-10"
      data-indeterminate={false}
      onChange={[Function]}
      type="checkbox"
    />
  </span>
</span>
`;
